SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  UserDetailsScreen.swift
+//  SwiftAppTemplate
+//
+//  Created by Vijay Goswami on 12/04/24.
+//
+
+import SwiftUI
+import Foundation
+
+struct UserDetailsScreen: View {
+    @State var name: String = ""
+    @State private var selectedGender: Gender = .male
+    @State private var dateOfBirth = Date()
+    @State private var showingDatePicker = false
+    @State var selectedCountry: String = "India"
+    var selectedLanguage: String {
+           guard let languageCode = Locale.current.languageCode,
+                 let languageName = Locale.current.localizedString(forLanguageCode: languageCode) else {
+               return "English"
+           }
+           return languageName
+       }
+    var startDate = Calendar.current.date(byAdding: .year, value: -100, to: Date()) ?? Date()
+    let onCompleted: () -> Void
+    
+    
+    var body: some View {
+        VStack {
+            Header(text: "UserDetails")
+            
+            nameField
```

Comment:
Same kind of fields also got used in edituserdetails screen is there a possibility of extracting them into reuseable structs?

Reply:
fixed